PyInstaller/utils/win32/icon.py

```diff
@@ -169,8 +169,15 @@ def splitter(s):
     else:
         logger.info("Updating icons from %s to %s", srcpath, dstpath)
 
+    try:
+        hsrc = win32api.LoadLibraryEx(srcpath, 0, LOAD_LIBRARY_AS_DATAFILE)
+    except:
+        # When srcpath is missing or points to an invalid file type,
+        # LoadLibraryEx returns a null handle and win32api raises an exception.
+        # Give the user some helpful info, and leave the icon resource unset.
+        logger.error("Unable to load icon file {}, ignoring --icon".format(srcpath))
```

Comment:
IMHO we should fail hard here. This should only happen if the user passed a wrong argument ti `--icon`, this this should not only be logged.

Reply:
Well, it was already failing hard in the sense of quitting with a traceback.

If I replace the logger.error() and return lines with, e.g., `raise ValueError("invalid icon file path {}"...)` then it would still quit with a traceback. Tracebacks are messy and look like a bug has happened. Is there another way to fail?

Reply:
`raise SystemExit(…)` and you are done :-) Short code, no traceback. (Given there is no other place is catching the SystemExit and printing the traceback. If there is, we should fix that other place).

Reply:
Please take a quick look around line 155 in that same module, where there are two errors related to finding icons in .ico files. Should those also be SystemExit? Right now they are ValueError.